analysis/active_analyses.R

```diff
@@ -0,0 +1,520 @@
+                                ipw = ipw, 
+                                strata = strata,
+                                covariate_sex = covariate_sex,
+                                covariate_age = covariate_age,
+                                covariate_other = gsub("cov_cat_smoking_status;","",all_covars), #-smoking status
+                                cox_start = cox_start,
+                                cox_stop = cox_stop,
+                                study_start = ifelse(c=="prevax", prevax_start, vax_unvax_start),
+                                study_stop = study_stop,
+                                cut_points = ifelse(c=="prevax", prevax_cuts, vax_unvax_cuts),
+                                controls_per_case = controls_per_case,
+                                total_event_threshold = total_event_threshold,
+                                episode_event_threshold = episode_event_threshold,
+                                covariate_threshold = covariate_threshold,
+                                age_spline = TRUE,
+                                analysis = "sub_smoking_ever")
+            
+            ## analysis: sub_smoking_current -----------------------------------------------
+            df[nrow(df)+1,] <- c(population = p,
+                                cohort = c,
+                                exposure = exposure, 
+                                outcome = i,
+                                ipw = ipw, 
+                                strata = strata,
+                                covariate_sex = covariate_sex,
+                                covariate_age = covariate_age,
+                                covariate_other = gsub("cov_cat_smoking_status;","",all_covars), #-smoking status
+                                cox_start = cox_start,
+                                cox_stop = cox_stop,
+                                study_start = ifelse(c=="prevax", prevax_start, vax_unvax_start),
+                                study_stop = study_stop,
+                                cut_points = ifelse(c=="prevax", prevax_cuts, vax_unvax_cuts),
+                                controls_per_case = controls_per_case,
+                                total_event_threshold = total_event_threshold,
+                                episode_event_threshold = episode_event_threshold,
+                                covariate_threshold = covariate_threshold,
+                                age_spline = TRUE,
+                                analysis = "sub_smoking_current")
+    
+        }
+
+    }
+
+}
+
+df$name <- paste0("cohort_",df$cohort, "-", 
+                  df$analysis, "-", 
+                  gsub("out_date_","",df$outcome), "-", 
+                  df$population)
+
+# Pre-existing populations only run the outcomes Pneumonia and Pulmonary fibrosis
+# Remove Pre-existing & asthma/copd rows
+df <- df[df$population != "preex_TRUE" | !df$outcome %in% c("out_date_asthma", "out_date_copd"),]
```

Comment:
This could be achieved using `outcomes_runall` and `outcomes_runmain` as for other projects if `preex_TRUE` was a subgroup rather than a population.

Reply:
As discussed, I have added the subgroup of preex_TRUE and preex_FALSE into loop and reflect that in the analysis coloum. So the above codes were deleted: 
https://github.com/opensafely/post-covid-respiratory/blob/ed683fbe43f338b0a9e37ad18f96623b4d43d216/analysis/active_analyses.R#L79-L109